probe-rs/targets/STM32WL_Series.yaml

```diff
@@ -2,32 +2,252 @@ name: STM32WL Series
   id: 0x20
   cc: 0x0
+generated_from_pack: true
+pack_file_release: 1.2.0
 variants:
-- name: STM32WLE5J8Ix
+- name: STM32WL54CCUx
   cores:
-  - name: main
+  - name: cm4
+    type: armv7em
+    core_access_options: !Arm
+      ap: 0
+      psel: 0x0
+  - name: cm0p
+    type: armv6m
+    core_access_options: !Arm
+      ap: 0
+      psel: 0x0
+  memory_map:
+  - !Ram
+    name: SRAM1
+    range:
+      start: 0x20000000
+      end: 0x20008000
+    cores:
+    - cm4
+    - cm0p
+  - !Ram
+    name: SRAM2
+    range:
+      start: 0x20008000
+      end: 0x20010000
+    cores:
+    - cm4
+    - cm0p
+  - !Nvm
+    name: FLASH
+    range:
+      start: 0x8000000
+      end: 0x8040000
+    cores:
+    - cm4
+    - cm0p
+    access:
+      write: false
+      boot: true
+  flash_algorithms:
+  - stm32wlxx_cm4
+  - stm32wlxx_cm0+
+- name: STM32WL54JCIx
+  cores:
+  - name: cm4
     type: armv7em
     core_access_options: !Arm
       ap: 0
       psel: 0x0
+  - name: cm0p
+    type: armv6m
+    core_access_options: !Arm
+      ap: 0
+      psel: 0x0
   memory_map:
   - !Ram
+    name: SRAM1
     range:
       start: 0x20000000
-      end: 0x20005000
+      end: 0x20008000
+    cores:
+    - cm4
+    - cm0p
+  - !Ram
+    name: SRAM2
+    range:
+      start: 0x20008000
+      end: 0x20010000
+    cores:
+    - cm4
+    - cm0p
+  - !Nvm
+    name: FLASH
+    range:
+      start: 0x8000000
+      end: 0x8040000
+    cores:
+    - cm4
+    - cm0p
+    access:
+      write: false
+      boot: true
+  flash_algorithms:
+  - stm32wlxx_cm4
+  - stm32wlxx_cm0+
+- name: STM32WL55CCUx
+  cores:
+  - name: cm4
+    type: armv7em
+    core_access_options: !Arm
+      ap: 0
+      psel: 0x0
+  - name: cm0p
+    type: armv6m
+    core_access_options: !Arm
+      ap: 0
+      psel: 0x0
+  memory_map:
+  - !Ram
+    name: SRAM1
+    range:
+      start: 0x20000000
+      end: 0x20008000
+    cores:
+    - cm4
+    - cm0p
+  - !Ram
+    name: SRAM2
+    range:
+      start: 0x20008000
+      end: 0x20010000
+    cores:
+    - cm4
+    - cm0p
+  - !Nvm
+    name: FLASH
+    range:
+      start: 0x8000000
+      end: 0x8040000
+    cores:
+    - cm4
+    - cm0p
+    access:
+      write: false
+      boot: true
+  flash_algorithms:
+  - stm32wlxx_cm4
+  - stm32wlxx_cm0+
+- name: STM32WL55JCIx
+  cores:
+  - name: cm4
+    type: armv7em
+    core_access_options: !Arm
+      ap: 0
+      psel: 0x0
+  - name: cm0p
+    type: armv6m
+    core_access_options: !Arm
+      ap: 0
+      psel: 0x0
+  memory_map:
+  - !Ram
+    name: SRAM1
+    range:
+      start: 0x20000000
+      end: 0x20008000
+    cores:
+    - cm4
+    - cm0p
+  - !Ram
+    name: SRAM2
+    range:
+      start: 0x20008000
+      end: 0x20010000
+    cores:
+    - cm4
+    - cm0p
+  - !Nvm
+    name: FLASH
+    range:
+      start: 0x8000000
+      end: 0x8040000
+    cores:
+    - cm4
+    - cm0p
+    access:
+      write: false
+      boot: true
+  flash_algorithms:
+  - stm32wlxx_cm4
+  - stm32wlxx_cm0+
```

Comment:
I get

```
FlashError(MultipleDefaultFlashLoaderAlgorithms { region: NvmRegion { name: Some("BANK_1"), range: 134217728..134479872, cores: ["cm4", "cm0p"], is_alias: false, access: Some(MemoryAccess { read: true, write: false, execute: true, boot: true }) } })
```

with this.

If I clear `default` on `stm32wlxx_cm0+` then `stm32wlxx_cm4` is chosen and flashing works.

If I use `stm32wlxx_cm0+` to flash, it leaves some sectors erased instead of programming them.

Reply:
> If I use stm32wlxx_cm0+ to flash, it leaves some sectors erased instead of programming them.

If a flash algo has issues, it's not something we can easily fix. Given the number of STM chips, I'd probably not even try, a pack update will just overwrite the modified algo anyway. The pack author may or may not know about such issues, and may or may not issue a fix.

Regardless, if a fix is as easy as removing an association, we can do that. However, I'd prefer if we could resolve the ambiguity instead.

Reply:
Well, if both algorithms worked, then I would suggest we have `FlashLoader` pick one instead of returning an error.  But since only one of them works for me, I was thinking about removing the other one from the four chips it's associated with, or removing its `default: true` bit.  Any preference?

Reply:
I think I would prefer the first one. If FlashLoader has multiple options to use (after narrowing down flash algos based on target address), it should prefer the algorithm that runs on the "main" core. We currently prefer the first core as the main one, and `Target` (the only user of `FlashLoader::new`) has this information.